src/mlpack/methods/ne/genome.hpp

```diff
@@ -0,0 +1,329 @@
+
+  // Links.
+  std::vector<LinkGene> aLinkGenes;
+
+  // Default constructor.
+  Genome() {}
+  
+  // Parametric constructor.
+  Genome(ssize_t id,
+  	     const std::vector<NeuronGene>& neuronGenes,
+         const std::vector<LinkGene>& linkGenes,
+         ssize_t numInput,
+         ssize_t numOutput,
+         double fitness,
+         double adjustedFitness):
+    aId(id),
+    aNeuronGenes(neuronGenes),
+    aLinkGenes(linkGenes),
+    aNumInput(numInput),
+    aNumOutput(numOutput),
+    aFitness(fitness),
+    aAdjustedFitness(adjustedFitness)
+  {}
+
+  // Copy constructor.
+  Genome(const Genome& genome) {
+    aId = genome.aId;
+    aNeuronGenes = genome.aNeuronGenes;
+    aLinkGenes = genome.aLinkGenes;
+    aNumInput = genome.aNumInput;
+    aNumOutput = genome.aNumOutput;
+    aFitness = genome.aFitness;
+    aAdjustedFitness = genome.aAdjustedFitness;
+  }
+
+  // Destructor.
+  ~Genome() {}
+
+  // Operator =.
+  Genome& operator =(const Genome& genome) {
+    if (this != &genome) {
+      aId = genome.aId;
+      aNeuronGenes = genome.aNeuronGenes;
+      aLinkGenes = genome.aLinkGenes;
+      aNumInput = genome.aNumInput;
+      aNumOutput = genome.aNumOutput;
+      aFitness = genome.aFitness;
+      aAdjustedFitness = genome.aAdjustedFitness;
+    }
+
+    return *this;
+  }
+
+  // Get genome id.
+  ssize_t Id() const { return aId; }
+
+  // Set genome id.
+  void Id(ssize_t id) { aId = id; }
+
+  // Get input length.
+  ssize_t NumInput() {
+    ssize_t numInput = 0;
+    for (ssize_t i=0; i<aNeuronGenes.size(); ++i) {
+      if (aNeuronGenes[i].Type() == INPUT || aNeuronGenes[i].Type() == BIAS)
+        ++numInput;
+    }
+    return numInput;
+  }
+
+  // Set input length.
+  void NumInput(ssize_t numInput) { aNumInput = numInput; }
+
+  // Get output length.
+  ssize_t NumOutput() {
+    ssize_t numOutput = 0;
+    for (ssize_t i=0; i<aNeuronGenes.size(); ++i) {  
+      if (aNeuronGenes[i].Type() == OUTPUT)
+        ++numOutput;
+    }
+    return numOutput;
+  }
+
+  // Set output length.
+  void NumOutput(ssize_t numOutput) { aNumOutput = numOutput; }
+
+  // Set fitness.
+  void Fitness(double fitness) { aFitness = fitness; }
+
+  // Get fitness.
+  double Fitness() const { return aFitness; }
+
+  // Set adjusted fitness.
+  void AdjustedFitness(double adjustedFitness) { aAdjustedFitness = adjustedFitness; }
+
+  // Get adjusted fitness.
+  double AdjustedFitness() const { return aAdjustedFitness; }
+
+  // Get neuron number.
+  ssize_t NumNeuron() const {
+    return aNeuronGenes.size();
+  }
+  
+  // Get link number.
+  ssize_t NumLink() const {
+    return aLinkGenes.size();
+  }
+
+  // Whether specified neuron id exist in this genome.
+  bool HasNeuronId(ssize_t id) const {
+    for (ssize_t i=0; i<NumNeuron(); ++i) {
+      if (aNeuronGenes[i].Id() == id) {
+        return true;
+      }
+    }
+    
+    return false;
+  }
+
+  // Get neuron by id.
+  NeuronGene GetNeuronById(ssize_t id) {
```

Comment:
We could avoid probably expensive copy (NeuronGene), by using a reference:

``void GetNeuronById(ssize_t id, NeuronGene& gene)``